commit cd407a5d21acca1685e4d3d965e1df50d60a54ec
Author: ZdenÄ›k Pavlas <zpavlas@redhat.com>
Date:   Wed Apr 4 12:29:53 2012 +0200

    Check .repo file permissions.  BZ 809598.
    
    Sometimes, people add .repo files with perm=600.
    Report the problem but don't abort.

diff --git a/yum/__init__.py b/yum/__init__.py
index 82a9380..c9bc815 100644
--- a/yum/__init__.py
+++ b/yum/__init__.py
@@ -542,6 +542,9 @@ class YumBase(depsolve.Depsolve):
 
             if os.path.isdir(reposdir):
                 for repofn in sorted(glob.glob('%s/*.repo' % reposdir)):
+                    if not os.access(repofn, os.R_OK):
+                        self.logger.warning(_("Skipping unreadable repository %s"), repr(repofn))
+                        continue
                     thisrepo_age = os.stat(repofn)[8]
                     if thisrepo_age < repo_config_age:
                         thisrepo_age = repo_config_age
